README.md

```diff
@@ -1,46 +1,66 @@
-
-Try running some of the following tasks:
-
-```shell
-npx hardhat accounts
-npx hardhat compile
-npx hardhat clean
-npx hardhat test
-npx hardhat node
-npx hardhat help
-REPORT_GAS=true npx hardhat test
-npx hardhat coverage
-npx hardhat run scripts/deploy.ts
-TS_NODE_FILES=true npx ts-node scripts/deploy.ts
-npx eslint '**/*.{js,ts}'
-npx eslint '**/*.{js,ts}' --fix
-npx prettier '**/*.{json,sol,md}' --check
-npx prettier '**/*.{json,sol,md}' --write
-npx solhint 'contracts/**/*.sol'
-npx solhint 'contracts/**/*.sol' --fix
+# About
+The Typescript library for developers to use with Medusa-enabled applications
+
+# Installation
+
+```bash
+npm install --save @medusa-network/medusa-sdk
 ```
 
-# Etherscan verification
+# Usage
 
-To try out Etherscan verification, you first need to deploy a contract to an Ethereum network that's supported by Etherscan, such as Ropsten.
+[See the demo application for more examples of usage](https://github.com/medusa-network/medusa-app)
 
-In this project, copy the .env.example file to a file named .env, and then edit it to fill in the details. Enter your Etherscan API key, your Ropsten node URL (eg from Alchemy), and the private key of the account which will send the deployment transaction. With a valid .env file in place, first deploy your contract:
+```typescript
+import { Medusa, EVMG1Point, SuiteType } from "@medusa-network/medusa-sdk";
 
-```shell
-hardhat run --network ropsten scripts/deploy.ts
-```
+// Initialize medusa with using the bn254 curve with the keypair in G1
+// using the HGamal encryption algorithm
+const medusa = await Medusa.init(SuiteType.BN254_KEYG1_HGAMAL);
+
+// Get Public Key of Medusa Oracle contract
+// Note: Medusa can be initialized one or more times for a given encryption suite
+const medusa = await Medusa.init(SuiteType.BN254_KEYG1_HGAMAL);
+const medusaContract = new ethers.Contract(
+  "0x..",
+  ["function distributedKey() external view returns (G1Point memory)"],
+  signerOrProvider
+);
+const medusaPublicKey = medusa.decodePublicKey(
+  await medusaContract.distributedKey()
+);
+
+// Encrypt data towards Medusa
+const { encryptedData, encryptedKey } = await medusa.encrypt(
+  "This is secret!",
+  medusaPublicKey,
+  myApplicationContractAddress,
+  userAddress
+);
 
-Then, copy the deployment address and paste it in to replace `DEPLOYED_CONTRACT_ADDRESS` in this command:
+// Derive a user's keypair from their (deterministic) signature
+const keypair = medusa.calculateKeypair(signature);
 
-```shell
-npx hardhat verify --network ropsten DEPLOYED_CONTRACT_ADDRESS "Hello, Hardhat!"
+// Decrypt response from Medusa
+const plaintext = await medusa.decrypt(
+  ciphertext,
+  encryptedContents,
+  keypair.secret,
+  medusaPublicKey
+);
+
+// Generate random keypair for testing purposes
+const medusa = await Medusa.init(SuiteType.BN254_KEYG1_HGAMAL);
+const keypair = Medusa.newKeypair(medusa.suite);
```

Comment:
why are these two lines helpful ?

Reply:
I left them as an example because newKeypair is a method in the public API. I figured it could be helpful for developers writing tests for their code that uses this sdk